### PR TITLE
change: adding check for main thread

### PR DESCRIPTION
**Pull request by krneta**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-simulator-v2-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/simulator_v2/base_simulator_v2.py [hunk 1]
```diff
@@ -245,6 +249,14 @@ def _validate_jaqcd(self, circuit_ir, qubit_count: int, shots: int):
             )
 
 
+def _validate_thread():
```
Comment (kshyatt-aws) [inline src/braket/simulator_v2/base_simulator_v2.py:252]: Can we add some kind of test for this warning?

**Review by kshyatt-aws — DISMISSED** [review]

**Review by rmshaffer — DISMISSED** [review]

File: src/braket/simulator_v2/base_simulator_v2.py (hunk 1 above)
Comment (krneta) [reply src/braket/simulator_v2/base_simulator_v2.py:252]: Done

**Review by rmshaffer — APPROVED** [review]

**Review by kshyatt-aws — APPROVED** [review]
LGTM. Just for my understanding (not blocking), do we have a suggested resolution for users? Is there something we can implement in the QBP to untangle this for them? Some env var they can set?

**Comment by krneta** [comment]
> LGTM. Just for my understanding (not blocking), do we have a suggested resolution for users? Is there something we can implement in the QBP to untangle this for them? Some env var they can set?

From what I can see, I think the changes would have to be made in Qiskit itself, or in juliacall.

**Pull request merged**